src/common/utils.h

```diff
@@ -374,6 +416,25 @@ inline void LogOnce(const std::string& message) {
   }
 }
 
+/*! \brief log storage fallback event
+ */
+inline void LogStorageFallback(const nnvm::NodeAttrs& attrs,
+                               const int dev_mask,
+                               const std::vector<int>* in_attrs,
+                               const std::vector<int>* out_attrs) {
+  static bool log = dmlc::GetEnv("MXNET_STORAGE_FALLBACK_LOG_VERBOSE", true);
+  if (!log) return;
+  const std::string op_str = operator_stype_string(attrs, dev_mask, *in_attrs, *out_attrs);
+  std::ostringstream os;
+  os << "\nStorage type fallback detected:\n" << op_str
```

Comment:
This might get called a lot, right? If so, do you need all.of the << calls or can the string be continued without it?

Reply:
Yeah I can use a multi-line string instead of all the ``<<``